core/src/main/scala/kafka/message/MessageSet.scala

```diff
@@ -54,8 +54,26 @@ object MessageSet {
    */
   def entrySize(message: Message): Int = LogOverhead + message.size
 
+  /**
+   * Validate the "magic" values of messages are the same in a compressed message set and return the max timestamp
+   * of the inner messages.
+   */
+  def validateMagicValuesAndGetTimestamp(messages: Seq[Message]): MagicAndTimestamp = {
```

Comment:
Would it be better to rename this to magicAndLargestTimestamp()? Also, the comment is outdated since we now return the magic as well.